compiler/resolution/functionResolution.cpp

```diff
@@ -603,8 +603,18 @@ const char* toString(CallInfo* info) {
 
 const char* toString(FnSymbol* fn) {
   if (fn->userString) {
-    if (developer)
-      return astr(fn->userString, " [", istr(fn->id), "]");
+    if (developer) {
+      const char* devStr;
+      devStr = astr(fn->userString);
+      if (fn->where) {
+        std::ostringstream oss;
+        oss << devStr << " where ";
+        fn->where->body.head->prettyPrint(&oss);
+        devStr = astr(oss.str().c_str());
+      }
+      devStr = astr(devStr, " [", istr(fn->id), "]");
+      return devStr;
+    }
```

Comment:
Looks like a good change, but unrelated to string-as-rec. I recommend doing this change on master now, to keep it separate from the string work.